### PR TITLE
Use unicode instead of string for writing files

### PR DESCRIPTION
**Pull request by ymjiang**
## Description ##
Fix the following error: 

```
File "~/machine_translation/dataprocessor.py", line 268, in write_sentences
    of.write(sent + '\n')
TypeError: must be unicode, not str
```

### CODE REVIEW

**Comment by ymjiang** [comment]
> @ymjiang thanks for the patch. Were you using python2 when running into the error?

Yes, python2

**Comment by szhengac** [comment]
It is strange that this error was not detected in the CI.

File: scripts/machine_translation/dataprocessor.py [hunk 1]
```diff
@@ -263,6 +263,6 @@ def write_sentences(sentences, file_path):
     with io.open(file_path, 'w', encoding='utf-8') as of:
         for sent in sentences:
             if isinstance(sent, (list, tuple)):
-                of.write(' '.join(sent) + '\n')
+                of.write(unicode(' '.join(sent) + '\n'))
```
Comment (leezu) [inline scripts/machine_translation/dataprocessor.py]: `unicode` is not defined in Py3. I don't think we need 'basic' strings here. If so, you may revert this change and add `from __future__ import unicode_literals` to the imports in the file

Comment (ymjiang) [reply scripts/machine_translation/dataprocessor.py]: I changed it in the patch

File: scripts/machine_translation/dataprocessor.py [hunk 2]
```diff
@@ -263,6 +263,6 @@ def write_sentences(sentences, file_path):
     with io.open(file_path, 'w', encoding='utf-8') as of:
         for sent in sentences:
             if isinstance(sent, (list, tuple)):
-                of.write(' '.join(sent) + '\n')
-            else:
-                of.write(sent + '\n')
+                of.write(u' '.join(sent) + u'\n')
+            else: 
+                of.write(u''.join(sent) + u'\n')
```
[the remark this answers is not in the document]
Comment (ymjiang) [reply scripts/machine_translation/dataprocessor.py]: Fixed. Sorry about this.

File: scripts/machine_translation/dataprocessor.py [hunk 3]
```diff
@@ -263,6 +263,6 @@ def write_sentences(sentences, file_path):
     with io.open(file_path, 'w', encoding='utf-8') as of:
         for sent in sentences:
             if isinstance(sent, (list, tuple)):
-                of.write(' '.join(sent) + '\n')
+                of.write(u' '.join(sent) + u'\n')
             else:
-                of.write(sent + '\n')
+                of.write(u' '.join(sent) + u'\n')
```
Comment (leezu) [inline scripts/machine_translation/dataprocessor.py]: This should be `of.write(sent + u'\n')`

**Comment by mli** [comment]
Job PR-541/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-541/8/index.html
